### PR TITLE
chore: DataStore docstrings

### PR DESCRIPTION
**Pull request by svidgen**
#### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/datastore/src/types.ts [hunk 1]
```diff
@@ -35,14 +35,31 @@ export type SchemaNamespaces = Record<string, SchemaNamespace>;
 export type SchemaNamespace = UserSchema & {
 	name: string;
 };
+
 export type SchemaModels = Record<string, SchemaModel>;
+
+/**
+ * something
```
Comment (dpilch) [inline packages/datastore/src/types.ts:42]: Haha, it certainly is something. Dropping a comment here so it's not forgotten.

File: packages/datastore/src/util.ts [hunk 2]
```diff
@@ -712,6 +869,12 @@ export class DeferredPromise {
 	}
 }
 
+/**
+ * Essentially a timeout for a callback.
+ *
+ * A callback is provided, which is called within `maxInterval` or when `resolve()`
+ * on the `DeferredCallbackResolver` is called &mdash; whichever happens first.
```
Comment (dpilch) [inline packages/datastore/src/util.ts:876]: Dash encoded for some reason.

**Review by dpilch — COMMENTED** [review]
I've made it half way through. So far really good information!

File: packages/datastore/src/util.ts (hunk 2 above)
Comment (svidgen) [reply packages/datastore/src/util.ts:876]: Yes. If you use markdown in docstrings, VSCode will mostly render it correctly on hover. I assume most docs site generators will do the same. Hence the entity encoding.

File: packages/datastore/src/types.ts (hunk 1 above)
Comment (svidgen) [reply packages/datastore/src/types.ts:42]: Oh man ... I think I might just strike some of the less complete docstrings for this pass. I want to try to tack my fix onto this PR, contrary to what I claimed I would do earlier this week.

Docstring burnout.

File: packages/core/src/Util/JobContext.ts [hunk 3]
```diff
@@ -0,0 +1,251 @@
+import Observable from 'zen-observable-ts';
+
+/**
+ * @private For internal Amplify use.
+ *
+ * A scope to collect background outstanding jobs, subscriptions, promises, and
```
Comment (svidgen) [inline packages/core/src/Util/JobContext.ts:6]: Note: Docstring is out of date, as it was speculative before TDDing. Don't read yet!

**Pull request closed without merging**